modules/scicos_blocks/src/c/cmat3d.c

```diff
@@ -180,6 +185,49 @@ SCICOS_BLOCKS_IMPEXP void cmat3d(scicos_block * block, scicos_flag flag)
             u = GetRealInPortPtrs(block, 1);
 
             result = pushData(block, u);
+
+            int iFigureUID;
+            int iAxeUID;
+            int iPlot3dUID;
+
+            int m, n, i, colormapLen, xMin, xMax, yMin, yMax, zMin, zMax, alpha, theta;
+
+            iFigureUID = getFigure(block);
+            iAxeUID = getAxe(iFigureUID, block);
+            iPlot3dUID = getPlot3d(iAxeUID, block);
+
+            m = GetInPortSize(block, 1, 1);
+            n = GetInPortSize(block, 1, 2);
+            colormapLen = block->ipar[3];
+            if (colormapLen == 1)
+            {
+                xMin = (double) 0;  // xMin
+                xMax = (double) m;  // xMax
+                yMin = (double) 0;  // yMin
+                yMax = (double) n;  // yMax
+            }
+            else
+            {
+                xMin = block->rpar[colormapLen + 0];   // xMin
+                xMax = block->rpar[colormapLen + 1];   // xMax
+                yMin = block->rpar[colormapLen + 2];   // yMin
+                yMax = block->rpar[colormapLen + 3];   // yMax
+            }
+
+            zMin = (double)block->ipar[0]; // zMin
+            zMax = (double)block->ipar[1]; // zMax
+
+            alpha = 50;      // alpha
+            theta = 280;     // theta
+            fprintf(filePointer, "%d %d || %d | %d | %d | %s | %d | %d || %d | %d | %d | %d | %d | %d | %d | %d ||",
```

Comment:
use %f for xMin, xMax, yMin, yMax, zMin, zMax, alpha, theta.
use %.0f to truncate values.
subtract 1 from minimum values before truncation.
add 1 to maximum values before truncation.